zip-proportional-output-fee-mechanism-pofm.rst

```diff
@@ -0,0 +1,242 @@
+shielded transaction, without consideration for the total number of inputs and outputs in the transaction.
+This has resulted in high output transactions with 1,100 outputs costing the same as 
+transactions with 2 outputs.
+
+
+Requirements for gathering consensus
+------------------------------------
+
+Wallet developers SHOULD update the fees to the proposed formula under Specification section
+by Madars Virza and Kris Nuttycombe [#madars-1]_ 
+
+Where #inputs and #outputs also take into account transparent inputs and outputs. 
+Otherwise, the fee structure (if not otherwise changed) will preferentially encourage 
+usage of the transparent part of the protocol.
+
+The change to the conventional transaction fees must be undertaken soon
+as the Zcash network has been under heavy load with high-output transactions while 
+regular shielded transactions with 2 outputs are affected when relying on current 
+light wallet infrastructure.
+
+The following parties need to be part of the consensus:
+
+* The technical aspects of a changing conventional fee based on outputs 
+need to be evaluated.
+* A guarantee from mining groups is required to include the updated POFM 
+transactions in the next block.
+* Wallet developers need to update the software to use the new fee.
+* Zcash documentation and community outreach must be undertaken to
+make the change known.
+
+
+Requirements for adoption
+-------------------------
+
+The change to the conventional transaction fees should be undertaken soon
+as it gets difficult to gain consensus with the growth in the network
+of wallets, exchanges, miners, and third parties involved.
+
+The following parties need to be part of the consensus:
+
+* Support from mining groups is required to include the updated conventional
+  fee transactions in the next block.
+* Wallet developers need to provide a commitment to update the software to use
+  the new fee.
+* Zcash documentation and community outreach must be undertaken to make the
+  change known.
+* Changes to librustzcash crates to:
+  * expose minimum recommended fee for a given number of recipients
+  * implement default fee equation and corresponding unit tests
+  * integration of such mechanism to Transaction Builder APIs
+* adopt `librustzcash` changes to mobile SDK repos:
+  * ZcashLightClientKit
+  * zcash-light-client-ffi
+  * zcash-android-wallet-sdk
+
+
+Specification
+=============
+
+Wallets implementing this specification will use a conventional fee in the form of 
+base_fee and marginal_fee.
+
+The proposal would end up selecting one of:
+* base_fee = 1000, marginal_fee = 250 in @nuttycom's proposal.
+* base_fee = 1000, marginal_fee = 1000 in @madars' proposal.
+* base_fee = 10000, marginal_fee = 2500 in @daira's proposal.
+* base_fee = 1000, marginal_fee = 1000 for Shielded, Shielding and De-shielding
+transactions, and base_fee = 10000, marginal_fee = 10000 for Transparent transactions 
```

Comment:
Does "Transparent transaction" here mean any transaction containing both transparent inputs and transparent outputs, irrespective of whether additional shielded inputs and outputs are present?

Reply:
Transparent transaction as in containing both transparent inputs and transparent outputs, irrespective of the shielded inputs and outputs being present. This is to incentivize shielded transaction use.

Reply:
The terms Shielded, Shielding, De-shielding, and Transparent might need to be defined in the "Terminology" section.